src/Simplex/Messaging/Server.hs

```diff
@@ -224,9 +234,10 @@ smpServer started cfg@ServerConfig {transports, transportConfig = tCfg} = do
           threadDelay' interval
 
     runClient :: Transport c => TProxy c -> c -> M ()
-    runClient _ h = do
+    runClient tp h = do
       kh <- asks serverIdentity
       smpVRange <- asks $ smpServerVRange . config
+      liftIO $ myThreadId >>= \tid -> labelThread tid ("smp handshake for " <> transportName tp)
```

Comment:
Point of interest: the thread is marked with handshake label and later gets re-marked with session ID.
This can distinguish "clients" that get stuck before entering session, something like the Slowloris attack.